src/shogun/classifier/AveragedPerceptron.cpp

```diff
@@ -9,6 +9,7 @@
 #include <shogun/features/iterators/DotIterator.h>
 #include <shogun/labels/Labels.h>
 #include <shogun/mathematics/linalg/LinalgNamespace.h>
+#include <shogun/lib/parameter_observers/ObservedValueTemplated.h>
```

Comment:
I wonder whether we could have some kind of meta include file that has a nicer name? `shogun/lib/observers/observers.h` (minor)
Also I suggest to rename that dir `parameter_observers` -> `observers` (even more minor)